spec/support/controller_helper.rb

```diff
@@ -19,6 +19,9 @@ def stub_sign_in(user = build(:user, password: VALID_PASSWORD))
     allow(controller).to receive(:user_session).and_return({}.with_indifferent_access)
     controller.auth_methods_session.authenticate!(TwoFactorAuthenticatable::AuthMethod::SMS)
     allow(controller).to receive(:current_user).and_return(user)
+    allow(controller).to receive(:sign_out) do
+      allow(controller).to receive(:current_user).and_return(nil)
+    end
```

Comment:
I think this change surfaced a real bug in the `logout_controller.rb` logic / specs. Before these changes, `current_user` will still produce a non-`nil` value in controller specs that use `stub_sign_in` after calling `sign_out`, despite the fact that real Devise code would start returning `nil`.

This is a problem for the order of how we call `current_user` after `sign_out` here, since `current_user` will always be `nil` in live code:

https://github.com/18F/identity-idp/blob/0e3107f9282a458d15c4c59a301b72e239d3b4ea/app/controllers/openid_connect/logout_controller.rb#L133-L139

This changed recently in #10573. Before that pull request, `redirect_user` was called before `sign_out`, so the actual user would have been passed correctly.

https://github.com/18F/identity-idp/blob/95e1cd18cb2c8049882650300cb7a73cbf9c823d/app/controllers/openid_connect/logout_controller.rb#L35-L36

The impact should be pretty limited, since the user is only used for consideration with the `openid_connect_redirect_uuid_override_map` configuration, which is used in test environments.

Reply:
Fixed in 0e91c51eaa.